LanguageFeatures/nnbd/weak/const_evaluation_A06_t01.dart

```diff
@@ -11,19 +11,15 @@
 
 // Requirements=nnbd-weak
 
+import "../../../Utils/expect.dart";
 import '../const_evaluation_lib.dart';
 import 'const_evaluation_legacy_lib.dart';
 
 const dynamic d = null;
 
 main() {
   const c1 = C.t(null, cLegacyInt);
```

Comment:
OK, tricky as usual. ;-)

The inferred type argument for `C` needs to be a supertype of `Null` and a supertype of `int*` (the type of `cLegacyInt`), but there are several types satisfying this. So we shouldn't have a compile-time error.

Next, no-sound-null-safety execution will not throw when `o as T` is tested in a situation where `o` evaluates to null. So we should indeed expect that there is no run-time error.

Very good, should work!